docs/eventing/samples/kafka/channel/README.md

```diff
@@ -151,3 +151,72 @@ Now you can see the events in the log of the `ksvc` using the command:
 ```
 kubectl logs --selector='serving.knative.dev/service=broker-kafka-display' -c user-container
 ```
+
+## Authentication against an Apache Kafka broker
+
+The KafkaSource supports TLS and SASL authentication methods.
+
+### TLS authentication
+
+For enabling TLS authentication, please have the below files
+
+* CA Certificate
+* Client Certificate and Key
+
+`KafkaChannel` expects these files to be in pem format, if it is in other format like jks, please convert to pem.
+
+
+1. Create the certificate files as secrets in the namespace of your choice, like:
+```
+$ kubectl create secret --namespace knative-eventing generic my-tls-secret \
+  --from-file=ca.crt=caroot.pem \
+  --from-file=user.crt=certificate.pem \
+  --from-file=user.key=key.pem
+ ```
+
+*NOTE:* It is important to use the same keys (`ca.crt`, `user.crt` and `user.key`).
+
+Now all you need is to reference your secret and its namespace on the `config-kafka` ConfigMap:
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-kafka
+  namespace: knative-eventing
+data:
+  #bootstrapServers: REPLACE_WITH_CLUSTER_URL
+  #authSecretName: name-of-your-secret-for-kafka-auth
+  #authSecretNamespace: namespace-of-your-secret-for-kafka-auth
+ ```
+
+### SASL authentication
+
+For enabling SASL authentication, please have the following data:
+
+* Username and password
+* Type of use SASL mechanism (e.g. `PLAIN`, `SCRAM-SHA-256` or `SCRAM-SHA-512`)
+
+1. Create the certificate files as secrets in the namespace of your choice, like:
+```
+$ kubectl create secret --namespace knative-eventing generic my-sasl-secret \
+  --from-file=ca.crt=caroot.pem \
+  --from-literal=password="SecretPassword" \
+  --from-literal=saslType="SCRAM-SHA-512" \
+  --from-literal=user="my-sasl-user"
+```
+
+*NOTE:* It is important to use the same keys (`user`, `password` and `saslType`). It's also recommended to enable TLS, in which case you need the `ca.crt` key as well! 
+
+Now all you need is to reference your secret and its namespace on the `config-kafka` ConfigMap:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-kafka
+  namespace: knative-eventing
+data:
+  #bootstrapServers: REPLACE_WITH_CLUSTER_URL
```

Comment:
Can we use a consistent format for variables please? Can we maybe format them all `<like-this>` as per the Kubernetes style guide? https://kubernetes.io/docs/contribute/style/style-guide/#use-angle-brackets-for-placeholders

Reply:
These still seem inconsistent, @matzew

Reply:
@abrennan89 I think this section is outdated (see the `Outdated` header above the code, next to the file name)

here is the actual file: https://github.com/knative/docs/pull/3067/files#diff-4259c7eeed66d201ebf025c842c68c0ad1a5c75121bde0869b80dc21ed7a14e8R216-R225